lib/auth/auth_with_roles.go

```diff
@@ -3410,10 +3416,38 @@ func (a *ServerWithRoles) CreateUser(ctx context.Context, user types.User) (type
 	if err := a.action(apidefaults.Namespace, types.KindUser, types.VerbCreate); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	if a.hasBuiltinOktaRole() {
+		if !hasOriginOkta(user) {
+			return nil, trace.BadParameter("Okta service must supply okta origin")
+		}
+	} else {
+		if hasOriginOkta(user) {
+			return nil, trace.BadParameter("Must be Okta service to set Okta origin")
```

Comment:
nit: these two `BadParameter` should be consistent
```suggestion
			return nil, trace.BadParameter("Must be Okta service to set okta origin")
```
may quote "okta"